tools/src/main/java/org/neo4j/tools/txlog/CheckTxLogs.java

```diff
@@ -94,20 +109,59 @@ public static void main( String[] args ) throws Exception
     // used in other projects do not remove!
     public boolean checkAll( File storeDirectory ) throws IOException
     {
-        return scan( storeDirectory, new PrintingInconsistenciesHandler( out ), CheckTypes.CHECK_TYPES );
+        PhysicalLogFiles logFiles = new PhysicalLogFiles( storeDirectory, fs );
+        InconsistenciesHandler handler = new PrintingInconsistenciesHandler( out );
+        boolean success = scan( logFiles, handler, CheckTypes.CHECK_TYPES );
+        success &= validateCheckPoints( logFiles, handler );
+        return success;
+    }
+
+    boolean validateCheckPoints( PhysicalLogFiles logFiles, InconsistenciesHandler handler ) throws IOException
+    {
+        long lowestLogVersion = logFiles.getLowestLogVersion();
+        Map<Long,Long> logFileSizes = new HashMap<>();
+        for ( long i = lowestLogVersion; i <= logFiles.getHighestLogVersion(); i++ )
```

Comment:
I would extract the `highestLogVersion` into a variable as well. And have that variable be (effectively) final. Mostly because it is good general practice for for-loops (to check against a constant value in the loop condition).

This is a good practice because the JIT prefers it that way, and even if it might not be important in this case, it is good practice to just always do it that way.